### ISSUE TITLE
Need to persist runner state

### ISSUE THREAD

**Issue by mapto** [relaypro-open/YarnRunner-Python#1]
Hello,
I'm participating in the [Interact-IF jam](https://interact-if.tumblr.com/post/672143584722649088/welcome-to-interact-ifs-2022-game-jam-the-jams). My team decided to use yarn, and I'm now trying to make a python web-based storytelling app.

My current challenge is that I need to preserve state across web requests. One naive option is to memorise the state as a sequence of player choices and re-run the narrative up to the current point. Clearly, this is suboptimal, even if the story is deterministic (not sure for how long this would be a given in a storytelling context).

A better solution would be to be able to initialise the runner with an optional state. Clearly this would not only be a node, but also variables, etc. I'd appreciate a comment on whether such an approach is consistent with your vision and wouldn't mind giving it a try aiming for a pull request.

Thanks a lot!
Martin

**Comment by weaversam8** [comment]
Yes, I see the value in something like this! I also agree with you- initializing with a state is better than passing a sequence of choices... There's no guarantee a story is non-deterministic. 

Open to a PR for sure! 

~~Also, I'm not sure if something like this exists in the C# or Unity APIs for Yarn, but if it does, we should do our best to mimic the pattern they use there.~~

**Edit:** after a quick search, I don't think anything like this exists in the Unity integration, beyond just controlling variable storage. We'll want to roll our own then. Probably a simple save / load paradigm.

**Comment by josefalanga** [comment]
Hello there @mapto. Do you have plans of continuing the work on the PR? Do you need help?

**Comment by mapto** [comment]
Hello! I'm sorry, but I am not going to work on this anymore.
